app/src/main/java/com/rud/movieapp/di/NetworkModule.kt

```diff
@@ -0,0 +1,53 @@
+package com.rud.movieapp.di
+
+import com.google.gson.Gson
+import com.rud.data.MovieWebService
+import com.rud.domain.Constants
+import com.rud.movieapp.BuildConfig
+import dagger.Module
+import dagger.Provides
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.gson.GsonConverterFactory
+import java.util.concurrent.TimeUnit
+
+@Module
+class NetworkModule {
+
+    @Provides
+    fun provideMovieApi(retrofit: Retrofit) = retrofit.create(MovieWebService::class.java)
```

Comment:
I think webservice should have singleton annotation.

Reply:
Fixed.